samples/getting-started/simple-algorithms/BernsteinVazirani.qs

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+
+// First, note that every Q# function must have a namespace. We define
+// a new one for this purpose.
+namespace Microsoft.Quantum.Samples.SimpleAlgorithms.BernsteinVazirani {
+    open Microsoft.Quantum.Intrinsic;
+    open Microsoft.Quantum.Canon;
+    open Microsoft.Quantum.Measurement;
+    open Microsoft.Quantum.Arrays;
+    open Microsoft.Quantum.Convert;
+
+    //////////////////////////////////////////////////////////////////////////
+    // Bernstein–Vazirani Fourier Sampling Quantum Algorithm //////////////////
+    //////////////////////////////////////////////////////////////////////////
+
+    /// # Summary
+    /// ParityViaFourierSampling implements the Bernstein-Vazirani quantum algorithm.
+    /// This algorithm computes for a given Boolean function that is promised to be
+    /// a parity 𝑓(𝑥₀, …, 𝑥ₙ₋₁) = Σᵢ 𝑟ᵢ 𝑥ᵢ a result in form of
+    /// a bit vector (𝑟₀, …, 𝑟ₙ₋₁) corresponding to the parity function.
+    /// Note that it is promised that the function is actually a parity function.
+    ///
+    /// # Input
+    /// ## Uf
+    /// A quantum operation that implements |𝑥〉|𝑦〉 ↦ |𝑥〉|𝑦 ⊕ 𝑓(𝑥)〉,
+    /// where 𝑓 is a Boolean function that implements a parity Σᵢ 𝑟ᵢ 𝑥ᵢ.
+    /// ## n
+    /// The number of bits of the input register |𝑥〉.
+    ///
+    /// # Output
+    /// An array of type `Bool[]` that contains the parity 𝑟⃗ = (𝑟₀, …, 𝑟ₙ₋₁).
+    ///
+    /// # See Also
+    /// - For details see Section 1.4.3 of Nielsen & Chuang.
+    ///
+    /// # References
+    /// - [ *Ethan Bernstein and Umesh Vazirani*,
+    ///     SIAM J. Comput., 26(5), 1411–1473, 1997 ](https://doi.org/10.1137/S0097539796300921)
+    operation ParityViaFourierSampling(Uf : ((Qubit[], Qubit) => Unit), n : Int) : Bool[] {
```

Comment:
I recognize that this is pre-existing, but it may be a good opportunity to rename this operation to use a verb instead of a noun (learn seems to align with descriptions of the B–V algorithm, perhaps?).